### PR TITLE
kdiff3: 1.7.x -> 1.8.1

### PR DESCRIPTION
**Pull request by dtzWill**
Drop git-diff patch, really hope it's not still outstanding
(and it doesn't apply).
cc @vcunat (who filed linked upstream issue)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/text/kdiff3/default.nix [hunk 1]
```diff
@@ -1,35 +1,18 @@
 {
-  mkDerivation, lib, fetchgit, fetchpatch,
+  mkDerivation, lib, fetchurl, # fetchgit, fetchpatch,
```
Comment (charles-dyfis-net) [inline pkgs/tools/text/kdiff3/default.nix]: Is there value to leaving in commented-out parameters?

File: pkgs/tools/text/kdiff3/default.nix [hunk 2]
```diff
@@ -1,35 +1,18 @@
...
-    sha256 = "0znlk9m844a6qsskbd898w4yk48dkg5bkqlkd5abvyrk1jipzyy8";
-    rev = "0d2ac328164e3cbe2db35875d3df3a86187ae84f";
+  src = fetchurl {
+    url = "https://download.kde.org/stable/${pname}/${pname}-${version}.tar.xz";
+    sha256 = "083pz5c1w7l9h4sb8zz8a763yph5sk3mxnhpdykz1rrggy9f8p54";
   };
 
-  setSourceRoot = ''sourceRoot="$(echo */kdiff3/)"'';
-
-  patches = [
```
Comment (charles-dyfis-net) [inline pkgs/tools/text/kdiff3/default.nix]: The steps needed to test whether this patch's functionality is included in the new release can be inferred by its source at https://gitlab.com/vcunat/kdiff3/commit/61061262161d90c015e8b53d5b3ba7df758d4bc0

Whereas the original patch just added stubs so kdiff3 wouldn't crash when given `--qall`/`--cs`/`--config`/`--confighelp`/`--L1`/`--L2`/`--L3`, the new one [is documented to actually support these options](https://github.com/KDE/kdiff3/blob/39d921959fe933007d79bb22b5409294420580f1/doc/en/index.docbook#L304-L317), confirming your conjecture that the patch is not only inapplicable but also unnecessary. :)

**Review by charles-dyfis-net — APPROVED** [review]
Looks good on a read; builds and executes on NixOS with sandboxing enabled.

File: pkgs/tools/text/kdiff3/default.nix (hunk 1 above)
Comment (dtzWill) [reply pkgs/tools/text/kdiff3/default.nix]: Oops, this was just a mistake :innocent:.  Will fix shortly.

File: pkgs/tools/text/kdiff3/default.nix (hunk 2 above)
Comment (dtzWill) [reply pkgs/tools/text/kdiff3/default.nix]: **Perfect**! Thanks for the great investigation and sharing what you found.  A+ would be reviewed again. ;)

**Comment by vcunat** [comment]
So far `git merge-tool` seems to work OK for me.

**Pull request merged**